pyatv/mrp/__init__.py

```diff
@@ -359,16 +367,21 @@ def __init__(self, loop, protocol, remote):
             protobuf.DEVICE_INFO_UPDATE_MESSAGE)
 
     async def _get_current_power_state(self):
-        if self._power_state is None:
-            current_power_state = await self.update_power_state(self.protocol.device_info, protobuf.DEVICE_INFO_MESSAGE)
-            return current_power_state
+        if self._current_power_state is None:
+            updated_power_state = await self.update_power_state(self.protocol.device_info, protobuf.DEVICE_INFO_MESSAGE)
+            return updated_power_state
         
-        return self._power_state
+        return self._current_power_state
 
     @property
-    def power_state(self):
+    def listener(self):
+        """Object receiving power state updates."""
+        return self.__listener
+
+    async def power_state(self):
```

Comment:
```suggestion
    def power_state(self):
```

Reply:
The only problem here. If I am changing it to non-async, I am getting the following error:

```
Traceback (most recent call last):
  File "/Volumes/Archive/Development/Github/pyatv/pyatv/__main__.py", line 572, in _run_application
    return await cli_handler(loop)
  File "/Volumes/Archive/Development/Github/pyatv/pyatv/__main__.py", line 390, in cli_handler
    return await _handle_commands(args, config, loop)
  File "/Volumes/Archive/Development/Github/pyatv/pyatv/__main__.py", line 482, in _handle_commands
    ret = await _handle_device_command(args, cmd, atv, loop)
  File "/Volumes/Archive/Development/Github/pyatv/pyatv/__main__.py", line 515, in _handle_device_command
    return (await _exec_command(
  File "/Volumes/Archive/Development/Github/pyatv/pyatv/__main__.py", line 538, in _exec_command
    value = await tmp(*args)
TypeError: object PowerState can't be used in 'await' expression

>>> An error occurred, full stack trace above
```

It seems that cli command cannot be non-async, which makes sense.